src/Device/deviceReducer.js

```diff
@@ -60,8 +63,13 @@ const initialState = {
 
 export default (state = initialState, action) => {
     switch (action.type) {
-        case DEVICES_DETECTED:
-            return { ...state, devices: [...action.devices] };
+        case DEVICES_DETECTED: {
+            const devices = action.devices.map(d => ({
+                ...d,
+                favorite: getIsFavoriteDevice(d.serialNumber),
```

Comment:
I'd suggest to add `nickname` to the device as well, it will help to use it for sorting and to avoid using _getDeviceNickname()_.
```
    nickname: getDeviceNickname(d.serialNumber),
```

Reply:
We have implemented this in the new version that we will push soon, but it wont update live so we are not able to use it

Reply:
I suspected that since there's no redux workflow to change the nickname.